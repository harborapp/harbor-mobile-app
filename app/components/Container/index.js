import React from 'react'
import PropTypes from 'prop-types'
import { View } from 'react-native'

/**
 * Container is a generic element to wrap children
 * in a flex View.
 *
 * @returns {jsx}
 */
const Container = ({ children, centerChildren, flex, style, ...props }) => {
  const styles = [
    { flex },
    style,
    centerChildren ? { alignItems: 'center', justifyContent: 'center' } : undefined
  ]

  return <View style={styles}>{children}</View>
}

Container.propTypes = {
  children: PropTypes.element.isRequired,
  style: View.propTypes.style,
  flex: PropTypes.number,
  centerChildren: PropTypes.bool
}

Container.defaultProps = {
  flex: 1,
  centerChildren: false
}

export default Container
