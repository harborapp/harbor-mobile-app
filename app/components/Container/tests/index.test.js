/* eslint-env jest */
import React from 'react'
import { Text } from 'react-native'
import renderer from 'react-test-renderer'

import Container from '../'

describe('Container', () => {
  it('should render', () => {
    const tree = renderer.create(
      <Container>
        <Text> Hello world </Text>
      </Container>
    )

    expect(tree).toMatchSnapshot()
  })
})
