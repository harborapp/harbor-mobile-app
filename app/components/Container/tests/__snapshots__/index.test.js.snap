// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container should render 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
      undefined,
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
     Hello world 
  </Text>
</View>
`;
